pandas/core/dtypes/cast.py

```diff
@@ -1918,7 +1917,13 @@ def np_can_hold_element(dtype: np.dtype, element: Any) -> Any:
                         # i.e. there are pd.NA elements
                         raise LossySetitemError
                 return element
+            # GH 57338
+            # Check boolean array set as object type
+            comp = [lib.is_bool(e) for e in np.array([element]).ravel()]
```

Comment:
I suspect this implementation can be very slow. What is this trying to achieve?

Reply:
Agreed. I'm trying to avoid sending a LossySetitemError when I want to cast an array of booleans that is set as an object type. So if I passed in ```array([True, False, True], dtype=object)``` to cast to a boolean dtype, the original code would raise a LossySetitemError. I basically just go through the array and check make sure each element is a boolean.